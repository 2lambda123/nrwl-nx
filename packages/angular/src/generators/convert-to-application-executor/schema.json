{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxAngularConvertToApplicationExecutorGenerator",
  "cli": "nx",
  "title": "Converts projects to use the `@nx/angular:application` executor or the `@angular-devkit/build-angular:application` builder.",
  "description": "Converts a project or all projects using one of the `@angular-devkit/build-angular:browser`, `@angular-devkit/build-angular:browser-esbuild`, `@nx/angular:browser` and `@nx/angular:browser-esbuild` executors to use the `@nx/angular:application` executor or the `@angular-devkit/build-angular:application` builder.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "Name of the Angular application project to convert. It has to contain a target using one of the `@angular-devkit/build-angular:browser`, `@angular-devkit/build-angular:browser-esbuild`, `@nx/angular:browser` and `@nx/angular:browser-esbuild` executors. If not specified, all projects with such targets will be converted.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-priority": "important"
    },
    "useNxExecutor": {
      "description": "Convert targets to use the `@nx/angular:application` executor instead of the `@angular-devkit/build-angular:application` builder.",
      "type": "boolean",
      "default": true
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    }
  },
  "additionalProperties": false
}
