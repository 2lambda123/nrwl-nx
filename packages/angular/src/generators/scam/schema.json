{
  "$schema": "http://json-schema.org/draft-07/schema",
  "$id": "SCAMGenerator",
  "cli": "nx",
  "title": "SCAM Generator Options Schema",
  "type": "object",
  "description": "Creates a new, generic component definition in the given or default project.",
  "additionalProperties": false,
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path at which to create the component file, relative to the current workspace. Default is a folder with the same name as the component in the project root.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?"
    },
    "displayBlock": {
      "description": "Specifies if the style will contain `:host { display: block; }`.",
      "type": "boolean",
      "default": false,
      "alias": "b"
    },
    "inlineStyle": {
      "description": "Include styles inline in the component.ts file. Only CSS styles can be included inline. By default, an external styles file is created and referenced in the component.ts file.",
      "type": "boolean",
      "default": false,
      "alias": "s"
    },
    "inlineTemplate": {
      "description": "Include template inline in the component.ts file. By default, an external template file is created and referenced in the component.ts file.",
      "type": "boolean",
      "default": false,
      "alias": "t"
    },
    "viewEncapsulation": {
      "description": "The view encapsulation strategy to use in the new component.",
      "enum": ["Emulated", "None", "ShadowDom"],
      "type": "string",
      "alias": "v"
    },
    "changeDetection": {
      "description": "The change detection strategy to use in the new component.",
      "enum": ["Default", "OnPush"],
      "type": "string",
      "default": "Default",
      "alias": "c"
    },
    "style": {
      "description": "The file extension or preprocessor to use for style files, or 'none' to skip generating the style file.",
      "type": "string",
      "default": "css",
      "enum": ["css", "scss", "sass", "less", "none"]
    },
    "skipTests": {
      "type": "boolean",
      "description": "Do not create \"spec.ts\" test files for the new component.",
      "default": false
    },
    "inlineScam": {
      "type": "boolean",
      "description": "Create the NgModule in the same file as the Component.",
      "default": true
    }
  },
  "required": ["name"]
}
