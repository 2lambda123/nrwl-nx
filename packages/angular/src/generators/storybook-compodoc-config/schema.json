{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxAngularStorybookCompodocConfig",
  "title": "Configure Compodoc for Storybook",
  "description": "Configure Compodoc for Angular projects with Storybook.",
  "type": "object",
  "cli": "nx",
  "examples": [
    {
      "command": "nx g @nrwl/angular:storybook-compodoc-config --name=my-lib",
      "description": "Congifure compodoc for Storybook for the `my-lib` Angular project."
    },
    {
      "command": "x g @nrwl/angular:storybook-compodoc-config --all",
      "description": "Congifure compodoc for Storybook for all Angular projects with Storybook."
    }
  ],
  "properties": {
    "name": {
      "type": "string",
      "aliases": ["project", "projectName"],
      "description": "Project for which to configure compodoc.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What's the name of the project for which you want to configure compodoc?",
      "x-dropdown": "projects"
    },
    "all": {
      "description": "Configure compodoc for all Angular projects with Storybook.",
      "type": "boolean",
      "default": false
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "additionalProperties": false,
  "required": [],
  "examplesFile": "../../../docs/storybook-compodoc-config-examples.md"
}
