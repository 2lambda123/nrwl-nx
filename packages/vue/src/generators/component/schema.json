{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "NxVueComponent",
  "title": "Create a Vue Component",
  "description": "Create a Vue Component for Nx.",
  "type": "object",
  "examples": [
    {
      "command": "nx g component my-component --project=mylib",
      "description": "Generate a component in the `mylib` library"
    },
    {
      "command": "nx g component my-component --project=mylib --classComponent",
      "description": "Generate a class component in the `mylib` library"
    }
  ],
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?",
      "x-priority": "important"
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the component?",
      "x-priority": "important"
    },
    "js": {
      "type": "boolean",
      "description": "Generate JavaScript files rather than TypeScript files.",
      "default": false
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create `spec.ts` test files for the new component.",
      "default": false,
      "x-priority": "internal"
    },
    "directory": {
      "type": "string",
      "description": "Create the component under this directory (can be nested).",
      "alias": "dir",
      "x-priority": "important"
    },
    "flat": {
      "type": "boolean",
      "description": "Create component at the source root rather than its own directory.",
      "default": false
    },
    "export": {
      "type": "boolean",
      "description": "When true, the component is exported from the project `index.ts` (if it exists).",
      "alias": "e",
      "default": false,
      "x-prompt": "Should this component be exported in the project?"
    },
    "pascalCaseFiles": {
      "type": "boolean",
      "description": "Use pascal case component file name (e.g. `App.tsx`).",
      "alias": "P",
      "default": false
    },
    "pascalCaseDirectory": {
      "type": "boolean",
      "description": "Use pascal case directory name (e.g. `App/App.tsx`).",
      "alias": "R",
      "default": false
    },
    "routing": {
      "type": "boolean",
      "description": "Generate a library with routes."
    },
    "fileName": {
      "type": "string",
      "description": "Create a component with this file name."
    },
    "inSourceTests": {
      "type": "boolean",
      "default": false,
      "description": "When using Vitest, separate spec files will not be generated and instead will be included within the source files. Read more on the Vitest docs site: https://vitest.dev/guide/in-source.html"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false,
      "x-priority": "internal"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "jest", "none"],
      "description": "Test runner to use for unit tests.",
      "x-prompt": "What unit test runner should be used?"
    }
  },
  "required": ["name", "project"]
}
