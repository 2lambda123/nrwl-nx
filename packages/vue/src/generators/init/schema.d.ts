export interface InitSchema {
  skipFormat?: boolean;
  skipPackageJson?: boolean;
  routing?: boolean;
  style?: 'css' | 'scss' | 'less' | 'none';
}
