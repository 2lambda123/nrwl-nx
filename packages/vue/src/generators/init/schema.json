{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxVueInit",
  "title": "Init Vue Plugin",
  "description": "Initialize a Vue Plugin.",
  "cli": "nx",
  "type": "object",
  "properties": {
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "skipPackageJson": {
      "description": "Do not add dependencies to `package.json`.",
      "type": "boolean",
      "default": false
    }
  },
  "required": []
}
