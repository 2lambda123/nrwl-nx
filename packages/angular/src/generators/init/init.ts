import { cypressInitGenerator } from '@nrwl/cypress';
import {
  addDependenciesToPackageJson,
  ensurePackage,
  formatFiles,
  GeneratorCallback,
  logger,
  readNxJson,
  runTasksInSerial,
  Tree,
  updateNxJson,
} from '@nrwl/devkit';
import { jestInitGenerator } from '@nrwl/jest';
import { Linter } from '@nrwl/linter';
import { initGenerator as jsInitGenerator } from '@nrwl/js';
import { E2eTestRunner, UnitTestRunner } from '../../utils/test-runners';
import {
  addDependenciesToPackageJsonIfDontExist,
  getInstalledAngularVersionInfo,
  getInstalledPackageVersion,
  versions,
} from '../utils/version-utils';
import type { PackageVersions } from '../../utils/backward-compatible-versions';
import { Schema } from './schema';

export async function angularInitGenerator(
  tree: Tree,
  rawOptions: Schema
): Promise<GeneratorCallback> {
  const installedAngularVersionInfo = getInstalledAngularVersionInfo(tree);
  const tasks: GeneratorCallback[] = [];
  const options = normalizeOptions(rawOptions);

  const pkgVersions = versions(tree);

  const peerDepsToInstall = [
    '@angular-devkit/core',
    '@angular-devkit/schematics',
  ];
  let devkitVersion: string;
  peerDepsToInstall.forEach((pkg) => {
    const packageVersion = getInstalledPackageVersion(tree, pkg);

    if (!packageVersion) {
      devkitVersion ??=
        getInstalledPackageVersion(tree, '@angular-devkit/build-angular') ??
        pkgVersions.angularDevkitVersion;

      try {
        ensurePackage(pkg, devkitVersion);
      } catch {
        // @schematics/angular cannot be required so this fails but this will still allow wrapping the schematic later on
      }

      if (!options.skipPackageJson) {
        tasks.push(
          addDependenciesToPackageJson(tree, {}, { [pkg]: devkitVersion })
        );
      }
    }
  });
  setDefaults(tree, options);

  const jsTask = await jsInitGenerator(tree, {
    ...options,
    tsConfigName: options.rootProject ? 'tsconfig.json' : 'tsconfig.base.json',
    js: false,
    skipFormat: true,
  });
  tasks.push(jsTask);

  if (!options.skipPackageJson) {
    tasks.push(updateDependencies(tree, pkgVersions));
  }
  const unitTestTask = await addUnitTestRunner(
    tree,
    options,
    pkgVersions.jestPresetAngularVersion
  );
  tasks.push(unitTestTask);
  const e2eTask = await addE2ETestRunner(tree, options);
  tasks.push(e2eTask);

  addGitIgnoreEntry(tree, '.angular');

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}

function normalizeOptions(options: Schema): Required<Schema> {
  return {
    e2eTestRunner: options.e2eTestRunner ?? E2eTestRunner.Cypress,
    linter: options.linter ?? Linter.EsLint,
    skipFormat: options.skipFormat ?? false,
    skipInstall: options.skipInstall ?? false,
    skipPackageJson: options.skipPackageJson ?? false,
    style: options.style ?? 'css',
    unitTestRunner: options.unitTestRunner ?? UnitTestRunner.Jest,
    rootProject: options.rootProject,
  };
}

function setDefaults(host: Tree, options: Schema) {
  const nxJson = readNxJson(host);

  nxJson.generators = nxJson.generators || {};
  nxJson.generators['@nrwl/angular:application'] = {
    style: options.style,
    linter: options.linter,
    unitTestRunner: options.unitTestRunner,
    e2eTestRunner: options.e2eTestRunner,
    ...(nxJson.generators['@nrwl/angular:application'] || {}),
  };
  nxJson.generators['@nrwl/angular:library'] = {
    linter: options.linter,
    unitTestRunner: options.unitTestRunner,
    ...(nxJson.generators['@nrwl/angular:library'] || {}),
  };
  nxJson.generators['@nrwl/angular:component'] = {
    style: options.style,
    ...(nxJson.generators['@nrwl/angular:component'] || {}),
  };

  updateNxJson(host, nxJson);
}

function updateDependencies(
  tree: Tree,
  versions: PackageVersions
): GeneratorCallback {
  const angularVersion =
    getInstalledPackageVersion(tree, '@angular/core') ??
    versions.angularVersion;
  const angularDevkitVersion =
    getInstalledPackageVersion(tree, '@angular-devkit/build-angular') ??
    versions.angularDevkitVersion;
  const rxjsVersion =
    getInstalledPackageVersion(tree, 'rxjs') ?? versions.rxjsVersion;
  const tsLibVersion =
    getInstalledPackageVersion(tree, 'tslib') ?? versions.tsLibVersion;
  const zoneJsVersion =
    getInstalledPackageVersion(tree, 'zone.js') ?? versions.zoneJsVersion;

  return addDependenciesToPackageJsonIfDontExist(
    tree,
    {
      '@angular/animations': angularVersion,
      '@angular/common': angularVersion,
      '@angular/compiler': angularVersion,
      '@angular/core': angularVersion,
      '@angular/forms': angularVersion,
      '@angular/platform-browser': angularVersion,
      '@angular/platform-browser-dynamic': angularVersion,
      '@angular/router': angularVersion,
      rxjs: rxjsVersion,
      tslib: tsLibVersion,
      'zone.js': zoneJsVersion,
    },
    {
      '@angular/cli': angularDevkitVersion,
      '@angular/compiler-cli': angularVersion,
      '@angular/language-service': angularVersion,
      '@angular-devkit/build-angular': angularDevkitVersion,
      '@schematics/angular': angularDevkitVersion,
    }
  );
}

async function addUnitTestRunner(
  tree: Tree,
  options: Schema,
  jestPresetAngularVersion: string
): Promise<GeneratorCallback> {
  switch (options.unitTestRunner) {
    case UnitTestRunner.Jest:
      if (!options.skipPackageJson) {
        addDependenciesToPackageJsonIfDontExist(
          tree,
          {},
          {
            'jest-preset-angular': jestPresetAngularVersion,
          }
        );
      }

      return jestInitGenerator(tree, {
        skipPackageJson: options.skipPackageJson,
      });
    default:
      return () => {};
  }
}

async function addE2ETestRunner(
  tree: Tree,
  options: Schema
): Promise<GeneratorCallback> {
  switch (options.e2eTestRunner) {
    case E2eTestRunner.Cypress:
      return cypressInitGenerator(tree, {
        skipPackageJson: options.skipPackageJson,
      });
    default:
      return () => {};
  }
}

function addGitIgnoreEntry(host: Tree, entry: string) {
  if (host.exists('.gitignore')) {
    let content = host.read('.gitignore', 'utf-8');
    if (/^\.angular$/gm.test(content)) {
      return;
    }

    content = `${content}\n${entry}\n`;
    host.write('.gitignore', content);
  } else {
    logger.warn(`Couldn't find .gitignore file to update`);
  }
}

export default angularInitGenerator;
