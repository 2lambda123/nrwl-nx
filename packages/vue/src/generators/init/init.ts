import {
  addDependenciesToPackageJson,
  formatFiles,
  GeneratorCallback,
  removeDependenciesFromPackageJson,
  runTasksInSerial,
  Tree,
} from '@nx/devkit';

import {
  lessVersion,
  nxVersion,
  sassVersion,
  vitePluginVueVersion,
  vueRouterVersion,
  vueTestUtilsVersion,
  vueTsconfigVersion,
  vueTscVersion,
  vueVersion,
} from '../../utils/versions';
import { InitSchema } from './schema';

function updateDependencies(host: Tree, schema: InitSchema) {
  const tasks: GeneratorCallback[] = [];

  tasks.push(removeDependenciesFromPackageJson(host, ['@nx/vue'], []));

  let dependencies: { [key: string]: string } = {
    vue: vueVersion,
  };
  let devDependencies: { [key: string]: string } = {
    '@nx/vue': nxVersion,
    '@vue/tsconfig': vueTsconfigVersion,
    '@vue/test-utils': vueTestUtilsVersion,
    '@vitejs/plugin-vue': vitePluginVueVersion,
    'vue-tsc': vueTscVersion,
  };

  if (schema.routing) {
    dependencies['vue-router'] = vueRouterVersion;
  }

  if (schema.style === 'scss') {
    devDependencies['sass'] = sassVersion;
  } else if (schema.style === 'less') {
    devDependencies['less'] = lessVersion;
  }

  tasks.push(addDependenciesToPackageJson(host, dependencies, devDependencies));

  return runTasksInSerial(...tasks);
}

export async function vueInitGenerator(host: Tree, schema: InitSchema) {
  let installTask: GeneratorCallback = () => {};
  if (!schema.skipPackageJson) {
    installTask = updateDependencies(host, schema);
  }

  if (!schema.skipFormat) {
    await formatFiles(host);
  }

  return installTask;
}

export default vueInitGenerator;
