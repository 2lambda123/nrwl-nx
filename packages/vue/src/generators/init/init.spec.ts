import { readJson, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { vueInitGenerator } from './init';
import { InitSchema } from './schema';

// TODO: more or different to be added here
describe('init', () => {
  let tree: Tree;
  let schema: InitSchema = {
    skipFormat: false,
    unitTestRunner: 'vitest',
    routing: true,
  };

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
  });

  it('should add vue dependencies', async () => {
    await vueInitGenerator(tree, schema);
    const packageJson = readJson(tree, 'package.json');
    expect(packageJson).toMatchSnapshot();
  });
});
