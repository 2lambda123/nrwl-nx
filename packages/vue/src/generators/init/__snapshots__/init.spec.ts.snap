// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init should add vue dependencies 1`] = `
{
  "dependencies": {
    "vue": "^3.3.4",
    "vue-router": "^4.2.4",
  },
  "devDependencies": {
    "@nx/vue": "0.0.1",
    "@vitejs/plugin-vue": "^4.5.0",
    "@vue/test-utils": "^2.4.1",
    "@vue/tsconfig": "^0.4.0",
    "vue-tsc": "^1.8.8",
  },
  "name": "@proj/source",
}
`;
